Add more info about getting a REST API token

nylen opened the pull request.

retrofox commented on Readme.md:
```diff
@@ -5,10 +5,21 @@ Compatible with Node.js and web browsers.
-Introduce the `wpcom` dependency in your `package.json`, and
-then initialize it with your API Token.
+Introduce the `wpcom` dependency into your `package.json`, and then initialize
+it with your API token.
```
Maybe we can add that `token` is optional and it's needed for actions that needs permissions (add/edit/delete a post, etc)

nylen replied:
Done in 2eb3fcc.

retrofox replied:
Sorry @nylen 
I meant just to say that `token` is an optional parameter:

```
... and then initialize it with your API token (optional).
```
The example must be very very simple.

nylen replied:
If we're going to say that the API token is optional, shouldn't we provide some guidance about which situations require a token and which ones do not?

retrofox replied:
Absolutely yes but in another section.

nylen replied:
OK, how about moving these 2 examples (one that requires permissions & one that does not) to the **Examples** section?

retrofox replied:
Yes, we have to think in useful sections. It could be examples page, or in the same API doc, etc

nylen commented:
@retrofox the diff was getting ugly so I rebased.  Changes based on our discussion:

- Added new **Authentication** section with links about OAuth2 tokens.  These links are things I did not know when I started using `wpcom.js`, so I think they will be helpful to other users too.

- Added two simple examples in **Examples** section: one that requires a token and one that does not.

Does that help address your concerns about sections?  If not, please let me know what you'd like me to change.

retrofox commented on Readme.md:
```diff
@@ -25,10 +26,26 @@ Include `dist/wpcom.js` in a `<script>` tag:
+public site is something anyone can do.
+
+If you do need a token, here are some links that will help you generate one:
+- [OAuth2 Authentication](https://developer.wordpress.com/docs/oauth2/)
```
Add ` to this link like the other ones

nylen replied:
The `code style` is for code: library names, functions, etc.  If you want these links to look consistent I'd rather remove the \` from every link there.

retrofox replied:
It looks different. https://cloudup.com/cpVuGB4h3mK
